infra/.gitignore

```diff
@@ -0,0 +1,4 @@
+!build/
```

Comment:
Rather than adding this complicated .gitignore, can we just make the root level .gitignore specify an absolute build path? i.e. set `/build/` with the leading slash. 

I haven't tested this so make sure this works.